tools/monitoring/api/types.go

```diff
@@ -32,7 +32,7 @@ type PrometheusSpec struct {
 
 type Agent interface {
 	Add(sp StatsAccessor, spec *AgentSpec) error
```

Comment:
Remove Add and Update methods and Use a single `CreateOrUpdate` method. This is break build in various places and we can fix those.